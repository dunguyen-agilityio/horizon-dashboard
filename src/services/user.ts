// Models
import { TUser } from '@/models/User';

// Services
import { apiClient } from './api';

// Constants
import { API_ENTITY } from '@/constants';

export const getTeamMember = () =>
  apiClient.get<TUser[]>(
    `${API_ENTITY.USERS}?limit=3&populate[0]=avatar&populate[1]=role`,
  );
