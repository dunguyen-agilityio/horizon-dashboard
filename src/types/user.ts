// Model
import { TImage } from '@/models/Image';
import { TUser } from '@/models/User';

export type Role = { name: string };

export type UserExtend = Omit<TUser, 'avatar' | 'fullName'> & {
  avatar?: TImage;
};
