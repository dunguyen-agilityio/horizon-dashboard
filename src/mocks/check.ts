import { TCheck } from '@/types/check';

export const MOCK_CHECKS: TCheck[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    quantity: 28,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    quantity: 30,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
