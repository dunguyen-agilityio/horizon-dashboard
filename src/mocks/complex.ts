import { Complex, ComplexStatus } from '@/types/complex';

export const MOCK_COMPLEXES: Complex[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    status: ComplexStatus.DISABLE,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    status: ComplexStatus.ERROR,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
