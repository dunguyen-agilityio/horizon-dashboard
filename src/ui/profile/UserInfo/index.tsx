// Components
import Image from 'next/image';
import { Avatar } from '@nextui-org/avatar';
import { Text } from '@/components';
import ImageWithBlur from '@/components/ImageWithBlur';

// Constants
import { PROFILE_IMAGES } from '@/constants/images';

// Types
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';

//Models
import { Role } from '@/models/User';

const { BACKGROUND } = PROFILE_IMAGES;

interface IUserInfoProps {
  avatar?: string;
  username: string;
  phoneNumber?: string;
  role: Role;
  postsTotal?: number;
  followersCount?: number;
  followingCount?: number;
}

const UserInfo = ({
  avatar,
  phoneNumber,
  username,
  role,
  postsTotal = 0,
  followersCount = 0,
  followingCount = 0,
}: IUserInfoProps) => (
  <div className="dark:bg-indigo bg-white p-[17px] rounded-md">
    <div className="w-full h-[131px] relative">
      <ImageWithBlur
        fill
        src={BACKGROUND}
        alt="background-profile"
        additionalClasses="object-cover rounded-2xl"
        sizes="(max-width: 768px) 315px,
              (min-width: 769px) and (max-width: 1024px) 396px,
              (min-width: 1025px) 518px"
      />
      <div className="w-full flex justify-center top-[85px] absolute">
        <Avatar
          src={avatar}
          ImgComponent={Image}
          alt="avatar-user"
          imgProps={{ width: 87, height: 87 }}
          className="w-[87px] h-[87px] border-4 dark:border-indigo border-white"
        />
      </div>
    </div>
    <div className="flex flex-col justify-center mt-14">
      <div className="">
        <Text className="text-[20px] text-center font-extrabold">
          {username}
        </Text>
        <Text className="text-center">{phoneNumber}</Text>
        <Text
          size={TEXT_SIZE.sm}
          variant={TEXT_VARIANT.SECONDARY}
          className="text-center"
        >
          {role.name}
        </Text>
      </div>
      <div className="flex justify-center gap-11 my-5">
        <div className="w-14 text-center">
          <Text size={TEXT_SIZE.extra} className="font-extrabold leading-6">
            {postsTotal}
          </Text>
          <Text
            size={TEXT_SIZE.sm}
            variant={TEXT_VARIANT.SECONDARY}
            className="leading-5"
          >
            Posts
          </Text>
        </div>
        <div className="w-14 text-center">
          <Text size={TEXT_SIZE.extra} className="font-extrabold leading-6">
            {followersCount}
          </Text>
          <Text
            size={TEXT_SIZE.sm}
            variant={TEXT_VARIANT.SECONDARY}
            className="leading-5"
          >
            Followers
          </Text>
        </div>
        <div className="w-14 text-center">
          <Text size={TEXT_SIZE.extra} className="font-extrabold leading-6">
            {followingCount}
          </Text>
          <Text
            size={TEXT_SIZE.sm}
            variant={TEXT_VARIANT.SECONDARY}
            className="leading-5"
          >
            Following
          </Text>
        </div>
      </div>
    </div>
  </div>
);

export default UserInfo;
