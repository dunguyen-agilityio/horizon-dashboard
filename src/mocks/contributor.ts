import { Contributor } from '@/models/Contributor';
import { MOCK_USERS } from './user';

export const MOCK_CONTRIBUTORS = [
  {
    ...MOCK_USERS[0],
    rating: 23,
    template: 'Abstract Colors',
    createdAt: new Date('2024-01-13T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[1],
    rating: 80,
    template: 'Swipe Circles',
    createdAt: new Date('2024-08-13T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[2],
    template: 'Colorful Heaven',
    createdAt: new Date('2024-08-11T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[3],
    template: '3D Cubes Art',
    avatar: undefined,
  },
] as Contributor[];
