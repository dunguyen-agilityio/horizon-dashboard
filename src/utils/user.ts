import { TUser, User } from '@/models/User';
import { UserResponse } from '@/types/contributor';
import { StrapiResponse } from '@/types/strapi';
import { formatStrapiModel } from './strapi';

export const formatUser = (data: TUser) => {
  const user = new User(data);

  return { ...user, fullName: user.username };
};

export const formatUserResponse = (
  data: StrapiResponse<UserResponse>,
): User => {
  const { avatar, role, ...rest } = formatStrapiModel(data);

  const { attributes: avatarData, id } = avatar?.data ?? {};

  const hasAvatar = avatarData && id;

  return formatUser({
    ...rest,
    role: { name: role?.attributes.name ?? '' },
    ...(hasAvatar && {
      avatar: { hash: avatarData.hash, url: avatarData.url },
    }),
  });
};
