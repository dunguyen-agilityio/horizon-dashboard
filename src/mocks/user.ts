import { TImage } from '@/models/Image';
import { TUser } from '@/models/User';
import { StrapiResponse } from '@/types/strapi';
import { UserResponse1 } from '@/utils/nft';

export const MOCK_AVATAR: TImage = {
  hash: 'image_de1e29b0b7',
  url: 'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1726630608/image_de1e29b0b7.png',
};

export const MOCK_USERS: TUser[] = [
  {
    id: '1',
    username: 'Adela Parkson',
    firstName: 'User 1',
    email: 'user1@gmail.com',
    role: { name: 'Creative Director' },
    createdAt: new Date('2024-08-17T07:25:04.188Z'),
    rating: 29,
    avatar: MOCK_AVATAR,
  },
  {
    id: '2',
    username: 'user2',
    firstName: 'User 2',
    email: 'user2@gmail.com',
    role: { name: 'Product Designer' },
    createdAt: new Date('2024-08-13T07:25:04.188Z'),
    rating: 72,
    avatar: MOCK_AVATAR,
  },
  {
    id: '3',
    username: 'user3',
    firstName: 'User 3',
    email: 'user3@gmail.com',
    role: { name: 'Product Designer' },
    createdAt: new Date('2024-01-13T07:25:04.188Z'),
    rating: 89,
    avatar: MOCK_AVATAR,
  },
  {
    id: '4',
    username: 'user4',
    firstName: 'User 4',
    email: 'user4@gmail.com',
    role: { name: 'Junior Graphic Designer' },
    createdAt: new Date('2024-08-18T07:25:04.188Z'),
    rating: 52,
    avatar: MOCK_AVATAR,
  },
];

export const MOCK_ROLE_RESPONSE: StrapiResponse<{ id: string; name: string }> =
  {
    id: '1',
    attributes: { name: 'Developer' },
  };

export const MOCK_IMAGE_RESPONSE: StrapiResponse<TImage> = {
  id: '1',
  attributes: {
    hash: 'image_de1e29b0b7',
    url: 'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1726630608/image_de1e29b0b7.png',
  },
};

export const MOCK_USERS_RESPONSE: StrapiResponse<UserResponse1>[] =
  MOCK_USERS.map(({ id, ...attributes }) => ({
    attributes: {
      ...attributes,
      avatar: { data: MOCK_IMAGE_RESPONSE },
      role: MOCK_ROLE_RESPONSE,
    },
    id,
  }));
