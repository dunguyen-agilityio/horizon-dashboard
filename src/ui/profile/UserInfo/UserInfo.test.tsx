import { render } from '@testing-library/react';

// Components
import UserInfo from '.';

// Mocks
import { MOCK_USERS } from '@/mocks/user';

const { avatar, userName, role, postsTotal, followers, following } =
  MOCK_USERS[0];

describe('UserInfo component', () => {
  it('UserInfo should render match snapshot', () => {
    const { container } = render(
      <UserInfo
        avatar={avatar}
        userName={userName}
        role={role}
        postsTotal={postsTotal}
        followers={followers}
        following={following}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('UserInfo should render correct alt and avatar user', () => {
    const { getByAltText } = render(
      <UserInfo
        avatar={avatar}
        userName={userName}
        role={role}
        postsTotal={postsTotal}
        followers={followers}
        following={following}
      />,
    );

    const avatarUser = getByAltText('avatar-user');
    expect(avatarUser).toBeInTheDocument();
  });

  it('UserInfo should render correct user information', () => {
    const { getByText } = render(
      <UserInfo
        avatar={avatar}
        userName={userName}
        role={role}
        postsTotal={postsTotal}
        followers={followers}
        following={following}
      />,
    );

    expect(getByText(userName)).toBeInTheDocument();
  });
});
