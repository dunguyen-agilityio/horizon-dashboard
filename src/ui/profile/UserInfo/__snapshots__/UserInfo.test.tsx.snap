// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component UserInfo should render match snapshot 1`] = `
<div>
  <div
    class="dark:bg-indigo bg-white p-[17px] rounded-md"
  >
    <div
      class="w-full h-[131px] relative"
    >
      <img
        alt="background-profile"
        class="object-cover rounded-2xl"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 768px) 315px, (min-width: 769px) and (max-width: 1024px) 396px, (min-width: 1025px) 518px"
        src="/_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=16&q=75 16w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=32&q=75 32w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=48&q=75 48w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=64&q=75 64w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=96&q=75 96w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=128&q=75 128w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=256&q=75 256w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=384&q=75 384w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=640&q=75 640w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=750&q=75 750w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=828&q=75 828w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=1080&q=75 1080w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=1200&q=75 1200w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=1920&q=75 1920w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=2048&q=75 2048w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="absolute top-[85px] left-[130px] md:left-[300px] lg:left-[430px] xl:left-[490px] 2xl:left-[200px]"
      >
        <span
          class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny bg-default text-default-foreground rounded-full w-[87px] h-[87px] border-4 dark:border-indigo border-white"
          tabindex="-1"
        >
          <img
            alt="avatar-user"
            class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
            data-nimg="1"
            decoding="async"
            height="87"
            loading="lazy"
            src="/_next/image?url=%2Fdefault-avatar.webp&w=256&q=75"
            srcset="/_next/image?url=%2Fdefault-avatar.webp&w=96&q=75 1x, /_next/image?url=%2Fdefault-avatar.webp&w=256&q=75 2x"
            style="color: transparent;"
            width="87"
          />
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-center mt-14"
    >
      <div
        class=""
      >
        <p
          class="text-primary dark:text-white leading-xl text-[20px] text-center font-extrabold"
        >
          Adela Parkson
        </p>
        <p
          class="text-secondary dark:text-secondary text-small font-medium text-center"
        >
          Creative Director
        </p>
      </div>
      <div
        class="flex justify-center gap-11 my-5"
      >
        <div
          class="w-14 text-center"
        >
          <p
            class="text-primary dark:text-white text-1.5xl leading-xl font-extrabold leading-6"
          >
            0
          </p>
          <p
            class="text-secondary dark:text-secondary text-small font-medium leading-5"
          >
            Posts
          </p>
        </div>
        <div
          class="w-14 text-center"
        >
          <p
            class="text-primary dark:text-white text-1.5xl leading-xl font-extrabold leading-6"
          >
            0
          </p>
          <p
            class="text-secondary dark:text-secondary text-small font-medium leading-5"
          >
            Followers
          </p>
        </div>
        <div
          class="w-14 text-center"
        >
          <p
            class="text-primary dark:text-white text-1.5xl leading-xl font-extrabold leading-6"
          >
            0
          </p>
          <p
            class="text-secondary dark:text-secondary text-small font-medium leading-5"
          >
            Following
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserInfo component UserInfo should render without avatar props 1`] = `
<div>
  <div
    class="dark:bg-indigo bg-white p-[17px] rounded-md"
  >
    <div
      class="w-full h-[131px] relative"
    >
      <img
        alt="background-profile"
        class="object-cover rounded-2xl"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 768px) 315px, (min-width: 769px) and (max-width: 1024px) 396px, (min-width: 1025px) 518px"
        src="/_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=16&q=75 16w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=32&q=75 32w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=48&q=75 48w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=64&q=75 64w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=96&q=75 96w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=128&q=75 128w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=256&q=75 256w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=384&q=75 384w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=640&q=75 640w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=750&q=75 750w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=828&q=75 828w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=1080&q=75 1080w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=1200&q=75 1200w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=1920&q=75 1920w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=2048&q=75 2048w, /_next/image?url=%2Fprofile%2Fbackground-profile.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="absolute top-[85px] left-[130px] md:left-[300px] lg:left-[430px] xl:left-[490px] 2xl:left-[200px]"
      >
        <span
          class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny bg-default text-default-foreground rounded-full w-[87px] h-[87px] border-4 dark:border-indigo border-white"
          tabindex="-1"
        >
          <span
            aria-label="avatar-user"
            class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center text-inherit w-full h-full"
            role="img"
          >
            <svg
              aria-hidden="true"
              fill="none"
              height="80%"
              role="presentation"
              viewBox="0 0 24 24"
              width="80%"
            >
              <path
                d="M12 2C9.38 2 7.25 4.13 7.25 6.75C7.25 9.32 9.26 11.4 11.88 11.49C11.96 11.48 12.04 11.48 12.1 11.49C12.12 11.49 12.13 11.49 12.15 11.49C12.16 11.49 12.16 11.49 12.17 11.49C14.73 11.4 16.74 9.32 16.75 6.75C16.75 4.13 14.62 2 12 2Z"
                fill="currentColor"
              />
              <path
                d="M17.0809 14.1489C14.2909 12.2889 9.74094 12.2889 6.93094 14.1489C5.66094 14.9989 4.96094 16.1489 4.96094 17.3789C4.96094 18.6089 5.66094 19.7489 6.92094 20.5889C8.32094 21.5289 10.1609 21.9989 12.0009 21.9989C13.8409 21.9989 15.6809 21.5289 17.0809 20.5889C18.3409 19.7389 19.0409 18.5989 19.0409 17.3589C19.0309 16.1289 18.3409 14.9889 17.0809 14.1489Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-center mt-14"
    >
      <div
        class=""
      >
        <p
          class="text-primary dark:text-white leading-xl text-[20px] text-center font-extrabold"
        >
          Adela Parkson
        </p>
        <p
          class="text-secondary dark:text-secondary text-small font-medium text-center"
        >
          Creative Director
        </p>
      </div>
      <div
        class="flex justify-center gap-11 my-5"
      >
        <div
          class="w-14 text-center"
        >
          <p
            class="text-primary dark:text-white text-1.5xl leading-xl font-extrabold leading-6"
          >
            0
          </p>
          <p
            class="text-secondary dark:text-secondary text-small font-medium leading-5"
          >
            Posts
          </p>
        </div>
        <div
          class="w-14 text-center"
        >
          <p
            class="text-primary dark:text-white text-1.5xl leading-xl font-extrabold leading-6"
          >
            0
          </p>
          <p
            class="text-secondary dark:text-secondary text-small font-medium leading-5"
          >
            Followers
          </p>
        </div>
        <div
          class="w-14 text-center"
        >
          <p
            class="text-primary dark:text-white text-1.5xl leading-xl font-extrabold leading-6"
          >
            0
          </p>
          <p
            class="text-secondary dark:text-secondary text-small font-medium leading-5"
          >
            Following
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
